src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts

```diff
@@ -467,127 +423,197 @@ export abstract class TextureZeroInitTest extends GPUTest {
     }
     this.queue.submit([commandEncoder.finish()]);
   }
+}
 
-  static generateParams(readMethods: ReadMethod[]) {
+const paramsBuilder = params()
+  .combine(
+    poptions('readMethod', [
+      ReadMethod.CopyToBuffer,
+      ReadMethod.CopyToTexture,
+      ReadMethod.Sample,
+      ReadMethod.DepthTest,
+      ReadMethod.StencilTest,
+    ])
+  )
+  .combine(poptions('format', kUncompressedTextureFormats))
+  .expand(({ format }) => {
+    const info = kUncompressedTextureFormatInfo[format];
+    let aspects: { aspect: GPUTextureAspect }[] = [];
+    if (info.depth && info.stencil) {
+      aspects = [{ aspect: 'depth-only' }, { aspect: 'stencil-only' }];
+      return aspects;
+    }
+    aspects = [{ aspect: 'all' }];
+    return aspects;
+  })
+  .combine(poptions('mipLevelCount', kMipLevelCounts))
+  .combine(poptions('sampleCount', kSampleCounts))
+  // Multisampled textures may only have one mip
+  .unless(({ sampleCount, mipLevelCount }) => sampleCount > 1 && mipLevelCount > 1)
+  .combine(poptions('uninitializeMethod', kUninitializeMethods))
+  .unless(
```

Comment:
Nit: these two `.unless`s only depend on readMethod and format, so can come immediately after both have been defined, near the top.

Reply:
Maybe this would help make the parameter generation marginally faster too?